### PR TITLE
Init repo with base functionality

### PR DESCRIPTION
**Pull request by Staubiii**
Already provides a test covered middleware.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/composer.lock
+/composer.phar
+/phpunit.xml
+/.phpunit.result.cache
+/phpunit.phar
+/config/Migrations/schema-dump-default.lock
+/vendor/
+/.idea/
```
Comment (ravage84) [inline .gitignore:8]: We can discuss in the future, whether we want to add _some_ PhpStorm IDE configuration files. But for now it's OK.

File: src/Controller/AppController.php
Comment (ravage84) [inline src/Controller/AppController.php:1]: I don't think we will need this one.

File: src/Plugin.php
Comment (ravage84) [inline src/Plugin.php:1]: I think the new convention fo plugin classes is that the name of the plugin should be prefixed before `Plugin`.

Let's check that and rename if so.

File: src/Plugin.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+     * Register application container services.
+     *
+     * @param \Cake\Core\ContainerInterface $container The Container to update.
+     * @return void
+     * @link https://book.cakephp.org/4/en/development/dependency-injection.html#dependency-injection
+     */
+    public function services(ContainerInterface $container): void
+    {
+        // Add your services here
+    }
```
Comment (ravage84) [inline src/Plugin.php]: Do we need to keep all these? I guess it's OK for now. But we should reconsider for a future cleanup.

File: tests/schema.sql
Comment (ravage84) [inline tests/schema.sql:1]: We do not need this one.

Let's remove.

**Review by ravage84 — CHANGES_REQUESTED** [review]
Looks good. Let's do some minor cleanup and then LGTM! 🚀

File: src/Plugin.php
Comment (Staubiii) [reply src/Plugin.php:1]: You're right: https://book.cakephp.org/4/en/plugins.html#plugin-objects 

I'll rename it.

**Review by ravage84 — APPROVED** [review]
🚂

**Pull request merged**